File: GeneratedEntityReaderClasses/Components/AreaDamageComponent.cs
namespace GeneratedNoitaClasses;
public class AreaDamageComponent : NoitaComponentBase,  iNoitaType<AreaDamageComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaCVector2<NoitaFloat> aabb_min { get; set; } = new();
	public NoitaCVector2<NoitaFloat> aabb_max { get; set; } = new();
	public NoitaFloat circle_radius { get; set; } = new();
	public NoitaUInt damage_type { get; set; } = new();
	public NoitaFloat damage_per_frame { get; set; } = new();
	public NoitaInt update_every_n_frame { get; set; } = new();
	public NoitaUInt entity_responsible { get; set; } = new();
	public NoitaString death_cause { get; set; } = new();
	public NoitaString entities_with_tag { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		aabb_min.Read(s);
		aabb_max.Read(s);
		circle_radius.Read(s);
		damage_type.Read(s);
		damage_per_frame.Read(s);
		update_every_n_frame.Read(s);
		entity_responsible.Read(s);
		death_cause.Read(s);
		entities_with_tag.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		aabb_min.Write(s);
		aabb_max.Write(s);
		circle_radius.Write(s);
		damage_type.Write(s);
		damage_per_frame.Write(s);
		update_every_n_frame.Write(s);
		entity_responsible.Write(s);
		death_cause.Write(s);
		entities_with_tag.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/AreaDamageComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaCVector2
from BasicNoitaTypes import NoitaFloat
from BasicNoitaTypes import NoitaUInt
from BasicNoitaTypes import NoitaInt

from NoitaStream import NoitaStream
@dataclass
class AreaDamageComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    aabb_min:NoitaCVector2 = field(default_factory=lambda: NoitaCVector2(NoitaFloat))
    aabb_max:NoitaCVector2 = field(default_factory=lambda: NoitaCVector2(NoitaFloat))
    circle_radius:NoitaFloat = field(default_factory=NoitaFloat)
    damage_type:NoitaUInt = field(default_factory=NoitaUInt)
    damage_per_frame:NoitaFloat = field(default_factory=NoitaFloat)
    update_every_n_frame:NoitaInt = field(default_factory=NoitaInt)
    entity_responsible:NoitaUInt = field(default_factory=NoitaUInt)
    death_cause:NoitaString = field(default_factory=NoitaString)
    entities_with_tag:NoitaString = field(default_factory=NoitaString)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.aabb_min.Read(s)
        self.aabb_max.Read(s)
        self.circle_radius.Read(s)
        self.damage_type.Read(s)
        self.damage_per_frame.Read(s)
        self.update_every_n_frame.Read(s)
        self.entity_responsible.Read(s)
        self.death_cause.Read(s)
        self.entities_with_tag.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.aabb_min.Write(s)
        self.aabb_max.Write(s)
        self.circle_radius.Write(s)
        self.damage_type.Write(s)
        self.damage_per_frame.Write(s)
        self.update_every_n_frame.Write(s)
        self.entity_responsible.Write(s)
        self.death_cause.Write(s)
        self.entities_with_tag.Write(s)

